src/teleop/gui/src/components/Amino.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+<div class="wrap">
+  <h3>Amino Test Controls</h3>
+  <div class="box1 heaters">
+    <ToggleButton id="heater" v-bind:currentState="heaters[siteIndex].intended" :labelEnableText="'Heater '+(site)+' Intended'" :labelDisableText="'Heater '+(site)+' Intended'" v-on:change="toggleHeater(siteIndex)"/>
+    <p v-bind:style="{color: heaters[siteIndex].color}">Thermistor {{site}}: {{heaters[siteIndex].temp.toFixed(2)}} C°</p>
+  </div>
+  <div class="box1 shutdown">
+    <ToggleButton id="autoshutdown" v-bind:currentState="autoShutdownIntended" :labelEnableText="'Auto Shutdown Intended'" :labelDisableText="'Auto Shutdown Intended'" v-on:change="sendAutoShutdownCmd(!autoShutdownIntended)"/>
+  </div>
+</div>  
+</template>
+
+<script>
+import ToggleButton from './ToggleButton.vue'
+import ROSLIB from 'roslib'
+
+export default {
+  data () {
+    return {
+      siteIndex: 0,
+      heaters: [
+        {
+          enabled: false,
+          intended: false,
+          temp: 0,
+          color: 'grey'
+        },
+        {
+          enabled: false,
+          intended: false,
+          temp: 0,
+          color: 'grey'
+        },
+        {
+          enabled: false,
+          intended: false,
+          temp: 0,
+          color: 'grey'
+        }
+      ],
+
+      autoShutdownEnabled: true,
+      autoShutdownIntended: true,
+
+      temp_sub: null
+    }
+  },
+
+  created: function(){
+    this.temp_sub = new ROSLIB.Topic({
+      ros : this.$ros,
+      name : 'science_data/temperatures',
+      messageType : 'mrover/ScienceTemperatures'
+    });
+
+    this.temp_sub.subscribe((msg) => {
+      // Callback for temp_sub
+      console.log(msg)
+      for(let i = 0; i < 3; i++){
+        this.heaters[i].temp = msg.temperatures[i]
+      }
+    });
+  },
+
+  props: {
+    site: {
+        type: String,
+        required: true,
+    }
+  },
+
+  components: {
+    ToggleButton,
+  },
+
+  methods: {
+    toggleHeater: function(id) {
+      this.heaters[id].intended = !this.heaters[id].intended
+      let toggleHeaterServ = new ROSLIB.Service({
+        ros : this.$ros,
+        name : 'change_heater_state',
+        serviceType : 'mrover/ChangeHeaterState'
+      });
+      let request = new ROSLIB.ServiceRequest({
+        device: id,
+        enable: this.heaters[id].intended
+      });
+      toggleHeaterServ.callService(request, (result) => {
+        if (!result) {
```

Comment:
ok i just want to clarify that the value that we send you will probably always be true. we will almost always return true for the initial request. however, we might want to create a second topic to publish data on our actual state (which we can figure out eventually). this will give data on real updates of what's going on with the heater.